### ISSUE TITLE
Update recommended rule set

### ISSUE THREAD

**Issue by MelSumner** [ember-template-lint/ember-template-lint#746]
## Primary objectives:

* Remove "style" related rules from the recommended set
* Add Octane related rules to recommended set
* Update list of A11Y rules in recommended set

### Add

* no-abstract-roles
* no-action
* no-curly-component-invocation
* no-extra-mut-helper-argument
* no-implicit-this
* no-obsolete-elements
* no-positive-tabindex
* no-unnecessary-component-helper
* require-iframe-title
* require-valid-alt-text

### Remove

* block-indentation
* img-alt-attributes
* linebreak-style
* no-unnecessary-concat
* quotes
* self-closing-void-elements

### Change Default Config

* link-rel-noopener -> `strict` config

**Comment by rwjblue** [comment]
After reviewing with @MelSumner, our current recommendations for a good "recommended" set that satisfies the objectives above are:

### Added

* no-abstract-roles
* no-action-modifiers
* no-curly-component-invocation
* no-element-event-actions
* no-extra-mut-helper-argument
* no-implicit-this
* no-obsolete-elements
* no-positive-tabindex
* no-unnecessary-component-helper
* require-iframe-title
* require-valid-alt-text

### Removed

* block-indentation
* img-alt-attributes
* linebreak-style
* no-unnecessary-concat
* quotes
* self-closing-void-elements

### Change Default Config

* link-rel-noopener -> `strict` config

**Comment by bmish** [comment]
Overall looks good but I have a few questions.

Regarding the additions:

1. My understanding is that the [no-action-modifiers](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-action-modifiers.md) and [no-element-event-actions](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-element-event-actions.md) rules are exact opposites of each other. So why are they both enabled?

Regarding the removals:

1. Are the style rules (`block-indentation`, `linebreak-style`, `quotes`) removed because prettier for handlebars will take the place of these? Because otherwise, I see a lot of value in these rules for enforcing consistent, clean, and nit-free code.

To recap the other removals:

1. `img-alt-attributes` is deprecated and replaced by `require-valid-alt-text`.
2. I assume `no-unnecessary-concat` is removed because of the discussion in #604.
3. I assume `self-closing-void-elements` is removed because of the discussion in #691.

**Comment by rwjblue** [comment]
> My understanding is that the no-action-modifiers  and no-element-event-actions  rules are exact opposites of each other. So why are they both enabled?

Don't use `action` in templates **at all**, use `{{on}}` 😸

> Are the style rules (block-indentation, linebreak-style, quotes) removed because prettier for handlebars will take the place of these?

Yes, this is precisely why they are removed. Just like with eslint's recommendations, our "recommended" set should **only** be about "correctness" and not "stylistic opinion".

I absolutely still expect folks to bring these rules themselves or use prettier to enforce the style. 

> img-alt-attributes is deprecated and replaced by require-valid-alt-text.

Confirm.

> I assume no-unnecessary-concat is removed because of the discussion in #604.

Confirm.

> I assume self-closing-void-elements is removed because of the discussion in 
#691.

Confirm.

**Comment by rwjblue** [comment]
> My understanding is that the no-action-modifiers and no-element-event-actions rules are exact opposites of each other. So why are they both enabled?

Maybe instead of doing this we should just make another rule (e.g. `no-action`)?

**Comment by bmish** [comment]
Thanks for the explanations!

> > My understanding is that the no-action-modifiers and no-element-event-actions rules are exact opposites of each other. So why are they both enabled?
> 
> Maybe instead of doing this we should just make another rule (e.g. `no-action`)?

Either we could create a new rule or just update the error messages and documentation for the existing rules to suggest `{{on}}`.

**Comment by lvegerano** [comment]
Why `no-element-event-actions`? Why `{{on 'click this.someFn}}` is recommended over `onclick={{fn this.someFn}}`?

**Comment by rwjblue** [comment]
@lvegerano The [rule documentation](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-element-event-actions.md#no-element-event-actions) has some info here, but a rough summary is that `onclick` has issues in SSR modes, doesn't allow for scenarios where `el.addEventListener` works but `el.onclick = function(){}` doesn't (e.g. `svg` elements, web components, etc).

**Comment by rwjblue** [comment]
Updated the description to match the changes in https://github.com/ember-template-lint/ember-template-lint/pull/858. Specifically, replace both `no-action-modifiers` and `no-element-event-action` with `no-action`.

**Comment by rwjblue** [comment]
Closed by #899
